OpenSim/Common/Test/testComponentInterface.cpp

```diff
@@ -1033,11 +1033,18 @@ void testInputOutputConnections()
 
     // do any other input/output connections
     foo1->updInput("input1").connect(bar->getOutput("PotentialEnergy"));
+
     // Must throw an exception since subState0 is not a state of internalSub.
-    ASSERT_THROW(OpenSim::Exception, 
-        foo2->updInput("input1").connect(world.getOutput("./internalSub/subState0")));
+    ASSERT_THROW(OpenSim::Exception,
+        foo2->updInput("input1").connect(world.getComponent("./internalSub").getOutput("subState0")));
```

Comment:
The leading period is handled when `getComponent()` calls on `traversePathToComponent()` (see [here](https://github.com/opensim-org/opensim-core/blob/master/OpenSim/Common/Component.h#L2041)). Looks like the "." is handled as it doesn't go into any of the `if` control statements, and then it just moves on to the next part of the path (i.e. internalSub).

Definitely will go back to buff up the exceptions.